frontend/packages/dev-console/src/utils/pipeline-augment.ts

```diff
@@ -143,3 +168,73 @@ export const augmentRunsToData = (
   }
   return data;
 };
+
+export enum runStatus {
+  Succeeded = 'Succeeded',
+  Failed = 'Failed',
+  Running = 'Running',
+  'In progress' = 'In Progress',
+  Pending = 'Pending',
+  FailedToStart = 'FailedToStart',
+  PipelineNotStarted = 'PipelineNotStarted',
+  Skipped = 'Skipped',
+  Cancelled = 'Cancelled',
+}
+
+export const getRunStatusColor = (status: string): StatusMessage => {
+  switch (status) {
+    case runStatus.Succeeded:
+      return { message: 'Succeded', pftoken: chart_color_green_400 };
+    case runStatus.Failed:
+      return { message: 'Failed', pftoken: global_danger_color_100 };
+    case runStatus.Running:
+      return { message: 'Running', pftoken: chart_color_blue_300 };
+    case runStatus['In Progress']:
+      return { message: 'Running', pftoken: chart_color_blue_300 };
+    case runStatus.Pending:
+      return { message: 'Not Started Yet', pftoken: chart_color_blue_100 };
+    case runStatus.FailedToStart:
+      return {
+        message: 'PipelineRun Failed To Start',
+        pftoken: global_danger_color_100,
+      };
+    case runStatus.Skipped:
+      return { message: 'Skipped', pftoken: chart_color_black_200 };
+    case runStatus.Cancelled:
+      return { message: 'Cancelled', pftoken: chart_color_black_400 };
+    default:
+      return { message: 'PipelineRun Not started Yet', pftoken: chart_color_black_200 };
+  }
+};
+
+export const getTaskStatus = (pipelinerun: PipelineRun): TaskStatus => {
+  let taskStatus;
+  Object.keys(runStatus).forEach((key) => {
+    taskStatus = { ...taskStatus, key };
```

Comment:
```
  const taskStatus: TaskStatus = {};
  Object.keys(runStatus).forEach((key) => {
    taskStatus[key] = 0;
  });
```

Reply:
this is throwing error, cannot set value `Success` (first key) of undefined